src/nvim/api/command.c

```diff
@@ -377,68 +382,96 @@ String nvim_cmd(uint64_t channel_id, Dict(cmd) *cmd, Dict(cmd_opts) *opts, Arena
     ea.argt = get_cmd_argt(ea.cmdidx);
   }
 
+  bool count_from_first_arg = false;
   // Parse command arguments since it's needed to get the command address type.
   if (HAS_KEY(cmd, cmd, args)) {
-    // Process all arguments. Convert non-String arguments to String and check if String arguments
-    // have non-whitespace characters.
-    args = arena_array(arena, cmd->args.size);
-    for (size_t i = 0; i < cmd->args.size; i++) {
-      Object elem = cmd->args.items[i];
-      char *data_str;
-
-      switch (elem.type) {
-      case kObjectTypeBoolean:
-        data_str = arena_alloc(arena, 2, false);
-        data_str[0] = elem.data.boolean ? '1' : '0';
-        data_str[1] = NUL;
-        ADD_C(args, CSTR_AS_OBJ(data_str));
+    if (cmd->args.size == 1 && (ea.argt & EX_COUNT) && !(ea.argt & EX_EXTRA)) {
+      Object first_arg = cmd->args.items[0];
+      bool is_numeric = false;
+      int64_t count_value = 0;
+
+      if (first_arg.type == kObjectTypeInteger) {
+        is_numeric = true;
+        count_value = first_arg.data.integer;
+      } else if (first_arg.type == kObjectTypeString) {
+        char *endptr;
+        long val = strtol(first_arg.data.string.data, &endptr, 10);
+        if (*endptr == '\0' && first_arg.data.string.size > 0) {
```

Comment:
need a hint comment that shows an example for each of these cases
```suggestion
        if (*endptr == '\0' && first_arg.data.string.size > 0) {
        // "..."
```